#!/usr/bin/env python

PACKAGE = "mur_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# ROLL
gen.add("p_r", double_t,   0, "Proportional parameter ROLL",       58, 0, 100)
gen.add("i_r", double_t,   0, "Integral parameter ROLL",           25, 0, 100)
gen.add("d_r", double_t,   0, "Derivative parameter ROLL",         32, 0, 100)

gen.add("roll_d", double_t,  0, "ROLL desire",     0.0, -4, 4)

# PITCH
gen.add("p_p", double_t,   0, "Proportional parameter PITCH",       58, 0, 100)
gen.add("i_p", double_t,   0, "Integral parameter PITCH",           25, 0, 100)
gen.add("d_p", double_t,   0, "Derivative parameter PITCH",         32, 0, 100)

gen.add("pitch_d", double_t,  0, "PITCH desire",     0.0, -4, 4)

# YAW
gen.add("p_yaw", double_t,   0, "Proportional parameter YAW",       58, 0, 100)
gen.add("i_yaw", double_t,   0, "Integral parameter YAW",           25, 0, 100)
gen.add("d_yaw", double_t,   0, "Derivative parameter YAW",         32, 0, 100)

gen.add("yaw_d", double_t,  0, "YAW desire",     0.0, -4, 4)

exit(gen.generate(PACKAGE, "mur_attitude_control", "MurAttitudeControl"))
